tests/features/load_configuration.feature

```diff
@@ -0,0 +1,43 @@
+Feature: Loading values into the configuration
+
+Scenario: Load configuration initialised from the Manifest
+    When I configure the app to run in the "skipBugsnag" state
+    And I run "LoadConfigurationFromManifestScenario"
+    Then I wait to receive a request
+    And the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier with the apiKey "abc12312312312312312312312312312"
+    And the exception "message" equals "LoadConfigurationFromManifestScenario"
+    And the event "app.releaseStage" equals "testing"
+    And the payload field "events.0.breadcrumbs" is an array with 1 elements
+    And the event "metaData.test.foo" equals "bar"
+    And the event "metaData.test.filter_me" equals "[REDACTED]"
+    And the event "app.versionCode" equals 753
+    #And the event "app.buildUUID" equals "test-7.5.3"
+    And the event "app.version" equals "7.5.3"
+    And the event "app.type" equals "test"
+    And the payload field "events.0.threads" is a non-empty array
+
+Scenario: Load configuration initialised with Kotlin
+    When I configure the app to run in the "skipBugsnag" state
+    And I run "LoadConfigurationKotlinScenario"
+    Then I wait to receive a request
+    And the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier with the apiKey "12312312312312312312312312312312"
+    And the exception "message" equals "LoadConfigurationKotlinScenario"
+    And the event "app.releaseStage" equals "testing"
+    And the payload field "events.0.breadcrumbs" is an array with 1 elements
+    And the event "metaData.test.foo" equals "bar"
+    And the event "metaData.test.filter_me" equals "[REDACTED]"
+    And the event "app.versionCode" equals 753
+    And the event "app.buildUUID" equals "test-7.5.3"
+    And the event "app.version" equals "7.5.3"
+    And the event "app.type" equals "test"
+    And the payload field "events.0.threads" is a non-empty array
+
+Scenario: Load configuration initialised with nulls
+    When I configure the app to run in the "skipBugsnag" state
+    And I run "LoadConfigurationNullsScenario"
+    Then I wait to receive a request
+    And the request is valid for the error reporting API version "4.0" for the "Android Bugsnag Notifier" notifier with the apiKey "12312312312312312312312312312312"
+    And the exception "message" equals "LoadConfigurationNullsScenario"
+    And the event "metaData.test.foo" equals "bar"
+    And the event "metaData.test.filter_me" equals "foobar"
```

Comment:
Need more expectations that fields are not present/null/set to defaults to reflect the setting of nulls in the config.

Reply:
I agree, however it's likely the behaviour will change with the `null` -> `no-op` update.
In that case, I can either set expectations at this point based on how I think it'll look, or wait until I know how it should work.

Reply:
Ok, that makes sense.  Could be a case for adding a "# TODO blah" comment in the meantime, just to guard a little against forgetting?